src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/Cas3UpdateDomainEventTypeForPersonDepartureUpdatedJobTest.kt

```diff
@@ -0,0 +1,69 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.integration
+
+import com.fasterxml.jackson.module.kotlin.readValue
+import org.assertj.core.api.Assertions
+import org.junit.jupiter.api.Test
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.events.cas3.model.CAS3PersonDepartureUpdatedEvent
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.events.cas3.model.EventType
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.model.MigrationJobType
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.factory.events.cas3.CAS3PersonDepartedEventDetailsFactory
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.DomainEventType
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.util.randomDateTimeBefore
+import java.time.Instant
+import java.util.UUID
+
+class Cas3UpdateDomainEventTypeForPersonDepartureUpdatedJobTest : MigrationJobTestBase() {
+
+  @Test
+  fun `update all person departure updated domain events when event type is person departure`() {
+    val personDepartureUpdatedInvalidDomainEvents = generateSequence {
+      domainEventFactory.produceAndPersist {
+        withType(DomainEventType.CAS3_PERSON_DEPARTURE_UPDATED)
+        withData(
+          objectMapper.writeValueAsString(
+            CAS3PersonDepartureUpdatedEvent(
+              id = UUID.randomUUID(),
+              timestamp = Instant.now().randomDateTimeBefore(),
+              eventType = EventType.personDeparted,
+              eventDetails = CAS3PersonDepartedEventDetailsFactory().produce(),
+            ),
+          ),
+        )
+      }
+    }.take(5).toList()
+
+    val personDepartureUpdatedValidDomainEvents = generateSequence {
+      domainEventFactory.produceAndPersist {
+        withType(DomainEventType.CAS3_PERSON_DEPARTURE_UPDATED)
+        withData(
+          objectMapper.writeValueAsString(
+            CAS3PersonDepartureUpdatedEvent(
+              id = UUID.randomUUID(),
+              timestamp = Instant.now().randomDateTimeBefore(),
+              eventType = EventType.personDepartureUpdated,
+              eventDetails = CAS3PersonDepartedEventDetailsFactory().produce(),
+            ),
+          ),
+        )
+      }
+    }.take(3).toList()
```

Comment:
can we please add a domain event of some other type and check it's unmodified after executing the seed job?

Reply:
Added another domain events